lib/libc/netdb/lib_freeaddrinfo.c

```diff
@@ -0,0 +1,68 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+
+#include <netdb.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: freeaddrinfo
+ *
+ * Description:
+ * Frees one or more addrinfo structures returned by getaddrinfo(), along with
+ * any additional storage associated with those structures. If the ai_next field
+ * of the structure is not null, the entire list of structures is freed.
+ *
+ * Input Parameters:
+ *   ai  - struct addrinfo to free
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: freeaddrinfo
+ ****************************************************************************/
+#ifdef CONFIG_NET_LWIP_NETDB
+void freeaddrinfo(FAR struct addrinfo *ai)
+{
+	DEBUGASSERT(ai != NULL);
+
+	struct lwip_info info_s;
+
+	int sockfd = socket(AF_INET, SOCK_DGRAM, 0);
```

Comment:
I think you'd better to set NETLIB_SOCK_IOCTL instead of SOCK_DGRAM  even they're same.
Because code reader can know that the socket is not for data transfering by this

Reply:
ok